indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -219,4 +269,75 @@ protected boolean isUseMaxMemoryEstimates()
         Tasks.DEFAULT_USE_MAX_MEMORY_ESTIMATES
     );
   }
+
+  protected ServiceMetricEvent.Builder getMetricBuilder()
+  {
+    return metricBuilder;
+  }
+
+  public IngestionMode getIngestionMode()
+  {
+    return ingestionMode;
+  }
+
+  protected static IngestionMode computeIngestionMode(@Nullable CompactionIOConfig ioConfig)
+  {
+    final boolean isAppendToExisting;
+    final boolean isDropExisting;
+
+    if (ioConfig == null) {
+      isAppendToExisting = BatchIOConfig.DEFAULT_APPEND_EXISTING;
+      isDropExisting = BatchIOConfig.DEFAULT_DROP_EXISTING;
+    } else {
+      isAppendToExisting = BatchIOConfig.DEFAULT_APPEND_EXISTING;
```

Comment:
How come this uses the default from BatchIOConfig instead of using ioConfig.isAppendToExisting()

Reply:
This is specific to `CompactionIOCOnfig` which does not have a `isAppendToExisting` flag.